### PR TITLE
Added .clang-format file and formatted everything

### PR DESCRIPTION
**Pull request by COM8**
This PR adds a `.clang-format` file based on the configuration specified in the Makefile.
I also run it over all files in the repo once.

### CODE REVIEW

File: external/bin/xxd.c [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /*
 As indicated at https://lists.debian.org/debian-legal/2015/01/msg00037.html,
-the author has permitted redistribution of xxd under the MIT license, as follows:
```
Comment (axsaucedo) [inline external/bin/xxd.c:3]: We should probably keep this file out of the scope of the formatter

File: python/src/main.cpp [hunk 2]
```diff
@@ -1,333 +1,557 @@
+#include <pybind11/numpy.h>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
-#include <pybind11/numpy.h>
```
Comment (axsaucedo) [inline python/src/main.cpp:3]: I'm not sure if this file shoudl also be encompassed by the formatter as it's very much structured to "circumvent" the shortcomings of the very cryptic ways in which pybind11 has to define the functions - mainly as adding the formatter could result in much less intuitive legibility on the already cryptic, so thinking we can skip this one as well

File: src/include/kompute/shaders/shaderlogisticregression.hpp [hunk 3]
```diff
@@ -24,408 +24,408 @@
 namespace kp {
```
Comment (axsaucedo) [inline src/include/kompute/shaders/shaderlogisticregression.hpp:24]: Also automatically generated so we shoudl skip

File: test/compiled_shaders_include/kompute_test/shaders/shadertest_logistic_regression.hpp [hunk 4]
```diff
@@ -23,411 +23,413 @@
 
 namespace kp {
 namespace shader_data {
-static const unsigned char test_shaders_glsl_test_logistic_regression_comp_spv[] = {
```
Comment (axsaucedo) [inline test/compiled_shaders_include/kompute_test/shaders/shadertest_logistic_regression.hpp:26]: Also automatically generated so we shoudl skip (all the compiled shared folders)

File: test/compiled_shaders_include/kompute_test/shaders/shadertest_op_custom_shader.hpp [hunk 5]
```diff
@@ -23,101 +23,103 @@
 
 namespace kp {
 namespace shader_data {
-static const unsigned char test_shaders_glsl_test_op_custom_shader_comp_spv[] = {
```
Comment (axsaucedo) [inline test/compiled_shaders_include/kompute_test/shaders/shadertest_op_custom_shader.hpp:26]: Also automatically generated so we shoudl skip (all the compiled shared folders)

File: vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.cpp [hunk 6]
```diff
@@ -15,782 +15,1111 @@
  */
 // This file is generated.
 #ifdef __cplusplus
-extern "C" {
```
Comment (axsaucedo) [inline vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.cpp:18]: This one as well (the vk_ndk_wrapper include)

File: vk_ndk_wrapper_include/kompute_vk_ndk_wrapper_patch.hpp [hunk 7]
```diff
@@ -2,86 +2,106 @@
 #pragma once
 
 /*
- * Vulkan API definitions to assist with issues in the `vulkan_wrapper.h/vulkan_wrapper.cpp`
```
Comment (axsaucedo) [inline vk_ndk_wrapper_include/kompute_vk_ndk_wrapper_patch.hpp:5]: This one as well (the vk_ndk_wrapper include)

File: vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.hpp [hunk 8]
```diff
@@ -18,397 +18,462 @@
 #define VULKAN_WRAPPER_H
 
 #ifdef __cplusplus
-extern "C" {
```
Comment (axsaucedo) [inline vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.hpp:21]: This one as well (the vk_ndk_wrapper include)

**Review by axsaucedo — CHANGES_REQUESTED** [review]
Great contribution - thank you! I added comments on the files that we should look to skip, but other than that looks good

File: python/src/main.cpp (hunk 2 above)
Comment (COM8) [reply python/src/main.cpp]: Yes, this one should not be formatted. Reverted it.

File: vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.cpp (hunk 6 above)
Comment (COM8) [reply vk_ndk_wrapper_include/kompute_vk_ndk_wrapper.cpp]: Done.

**Comment by COM8** [comment]
Those are things I could have thought about :D
Fixed it.

**Comment by axsaucedo** [comment]
Nice one! Thanks for updating it so fast - just having a look. I can see that the format command now reflects the `file` command, but I can see that also the `examples` folder also has some udpates - did you run the command through the `Makefile`? Or did you run it through a different command? Just thinking it makes sense that we also format the examples but may be worth making the command in the `makefile` consistent with the changes

**Comment by COM8** [comment]
I've used this script: https://github.com/eklitzke/clang-format-all/blob/master/clang-format-all
But you are right it makes sense to reflect the changes in the Makefile.
I fixed that.

**Comment by axsaucedo** [comment]
Perfect - looks great. Nice one @COM8 , merging - thanks for the contribution

**Pull request merged**